### ISSUE TITLE
Ensure appropriate labels are provided for color/gradient picker controls

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#54055]
## What problem does this address?
Changes introduced in #52255 make it a requirement for the [`CircularOptionPicker`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/circular-option-picker) to have an `aria-label`/`-labelledby` prop provided, to make sure appropriate context is available for the control. The three consuming components ([`ColorPalette`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/color-palette), [`DuotonePicker`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/duotone-picker) and [`GradientPicker`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/gradient-picker)) all hide this requirement behind a default value, and should transition away from optional ARIA props to requiring a label of some sort.

## What is your proposed solution?
All of the higher-order controls that use these three foundational components should be updated to provide a more appropriate and relevant label for their context, passing either an `aria-label` or `aria-labelledby` prop.

## Acceptance Criteria

 - [x] Identify all of the controls that use `ColorPalette`, `DuotonePicker` and `GradientPicker`
 - [ ] Update them to provide context-appropriate labels with `aria-label` or `aria-labelledby`
 - [ ] Update `ColorPalette`, `DuotonePicker` and `GradientPicker` to require one of `aria-label` or `aria-labelledby` (see [related conversation](https://github.com/WordPress/gutenberg/pull/52255#discussion_r1311711606))

**Comment by andrewhayward** [comment]
## Components that use `<ColorPalette>`:

 - [ ] [Block Editor `ColorGradient`](/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/colors-gradients/control.js#L73)
 - [ ] [Block Library `CoverEdit`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/cover/edit/index.js#L374)
 - [ ] [Block Library native `CoverEdit`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/cover/edit.native.js#L533)
 - [ ] [Components `BorderControl/BorderControlDropdown`](/WordPress/gutenberg/blob/trunk/packages/components/src/border-control/border-control-dropdown/component.tsx#L208)
 - [x] [Components `DuotonePicker/ColorListPicker`](/WordPress/gutenberg/blob/trunk/packages/components/src/duotone-picker/color-list-picker/index.tsx#L45)
   #54468
 - [ ] [Components native `mobile/ColorSettings`](/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/color-settings/palette.screen.native.js#L193)
 - [ ] [Components `PaletteEdit`](/WordPress/gutenberg/blob/trunk/packages/components/src/palette-edit/index.tsx#L621)
 - [ ] [Format Library `TextColor/ColorPicker`](/WordPress/gutenberg/blob/trunk/packages/format-library/src/text-color/inline.js#L126)

**Comment by andrewhayward** [comment]
## Components that use `<DuotonePicker>`:

 - [x] [Block Editor `DuotoneControl`](/WordPress/gutenberg/blob/trunk/packages/format-library/src/text-color/inline.js#L126)
       #54473
 - [ ] [Block Editor `GlobalStyles/FiltersPanel`](/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/global-styles/filters-panel.js#L210)
 - [ ] [Edit Site `GlobalStyles/GradientPalettePanel`](/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/gradients-palette-panel.js#L115)

**Comment by andrewhayward** [comment]
## Components that use `<GradientPicker>`:

 - [ ] [Block Editor `ColorsGradients/ColorGradientControl`](/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/colors-gradients/control.js#L93)
 - [ ] [Components `PaletteEdit`](/WordPress/gutenberg/blob/trunk/packages/components/src/palette-edit/index.tsx#L613)
